### PR TITLE
feat(vue-app): support async watchQuery

### PR DESCRIPTION
**Pull request by rwaness**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

It may be usefull to do async work inside the watchQuery option instead of rerun ALL the fetch option.

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-app/template/client.js [hunk 1]
```diff
@@ -157,7 +157,7 @@ async function loadAsyncComponents (to, from, next) {
         (Component, instance) => ({ Component, instance })
       )
       // Add a marker on each component that it needs to refresh or not
-      const startLoader = Components.some(({ Component, instance }) => {
+      const startLoader = Components.some(async ({ Component, instance }) => {
```
Comment (pi0) [inline packages/vue-app/template/client.js:160]: [Array.prototype.sum](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some#:~:text=Description,some()%20immediately%20returns%20true%20.) expects callback returning boolean not promise. We may use `map` + `Promise.all` + `some` if want to support this :)

Comment (rwaness) [reply packages/vue-app/template/client.js:160]: i'll do

**Comment by Atinux** [comment]
Thank you for the PR.

I am not sure to understand the need here, what would be the use case for this?

**Comment by rwaness** [comment]
Salut :)

ie: fetch do multiple async call to init the page dependings on multiple query string parameters
then some of thoses params change so I would like to rerun only the call linked to this param and not all the fetch stack ...

with async watchQuery, we may do async call and then return false to prevent all fetch and asyncData content to be fired again

imo it's to prevent the loading state and unecessary calls to server (fetch > store actions > api)
and so improve ux

I stay here if needed

**Review by clarkdo — COMMENTED** [review]
Nitpicking: With current change, every watchQuery will be executed even if one of first components returns true.

**Comment by rwaness** [comment]
@clarkdo it may be interested to fire fetch & async as soon as 1 watchQuery return true but all watchQuery have to run (even if 1 return true) because all async process in watchQuery have to be executed.

**Comment by pi0** [comment]
Hi @rwaness. Thanks for your contribution. After talking with the team, introducing this feature would be risky for regressions also making it different with other vue watchers that are not async.

You can maybe use a method to be shared with both fetch and watchQuery to do partial data fetching.

**Pull request closed without merging**